### PR TITLE
Feature/6839 support video files

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: app/models/concerns/has_file.rb [hunk 1]
```diff
@@ -11,13 +11,29 @@ def file_mandatory?
...
+  def file_path
+    self.image_path
+  end
+
+  def embed_path
+    self.image_path('embed')
+  end
+
   module ClassMethods
     def max_size
```
Comment (caiosba) [inline app/models/concerns/has_file.rb]: This method is confusing... can we implement one in `UploadedImage` and another one in `UploadedVideo`?

File: app/uploaders/image_uploader.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class ImageUploader < FileUploader
     process resize_to_fit: CONFIG['image_embed_size'] || [800, 600]
   end
 
-  def default_url
+  def default_url(*args)
```
Comment (caiosba) [inline app/uploaders/image_uploader.rb]: If we don't need `*args`, please remove it.

File: db/migrate/20191015080931_add_temp_location_for_media_file.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddTempLocationForMediaFile < ActiveRecord::Migration
+  def change
+  	add_column :medias, :file_tmp, :string
```
Comment (caiosba) [inline db/migrate/20191015080931_add_temp_location_for_media_file.rb]: What is this column used for?

File: app/models/concerns/has_file.rb [hunk 4]
```diff
@@ -11,13 +11,29 @@ def file_mandatory?
...
-      ENV['MAX_UPLOAD_SIZE'] ? Filesize.from("#{ENV['MAX_UPLOAD_SIZE']}B").to_f : (CONFIG['uploaded_file_max_size'] || 1.megabyte)
+      options = { var_name: 'MAX_UPLOAD_SIZE', config_name: 'uploaded_file_max_size', size: 1.megabyte}
+      if (self.name == 'UploadedVideo')
+        options = { var_name: 'MAX_VIDEO_SIZE', config_name: 'video_file_max_size', size: 20.megabyte }
+      end
+      ENV[options[:var_name]] ? Filesize.from("#{ENV[options[:var_name]]}B").to_f : (CONFIG[options[:config_name]] || options[:size])
```
Comment (infojunkie) [inline app/models/concerns/has_file.rb]: Plz follow the instructions given here to ignore this issue.

File: db/migrate/20191015080931_add_temp_location_for_media_file.rb (hunk 3 above)
Comment (melsawy) [reply db/migrate/20191015080931_add_temp_location_for_media_file.rb]: no need to this column as I used `process_in_background` option

File: app/models/bot/smooch.rb [hunk 5]
```diff
@@ -678,17 +681,19 @@ def self.save_text_message(message)
     end
   end
 
-  def self.save_image_message(message)
+  def self.save_image_message(message, type='image')
```
[the remark this answers is not in the document]
Comment (melsawy) [reply app/models/bot/smooch.rb]: Done

File: app/models/concerns/has_file.rb (hunk 1 above)
Comment (melsawy) [reply app/models/concerns/has_file.rb]: Done

File: app/uploaders/image_uploader.rb (hunk 2 above)
Comment (melsawy) [reply app/uploaders/image_uploader.rb]: Done

File: app/models/concerns/has_file.rb (hunk 4 above)
Comment (melsawy) [reply app/models/concerns/has_file.rb]: Done

**Pull request merged**